src/index.py

```diff
@@ -269,35 +304,39 @@ def _check_contents_item(
         )
 
     # Check whether item is hidden and a directory
-    item_path = docs_path / Path(item.reference_value)
-    if item.hidden and item_path.is_dir():
-        raise InputError(f"A hidden item is a directory. {item=!r}")
+    item_reference_type = classify_item_reference(
+        reference=item.reference_value, docs_path=docs_path
+    )
 
-    # Check that the next item is within the directory
-    item_relative_path = Path(item.reference_value)
-    try:
-        item_to_aggregate_path = item_relative_path.relative_to(aggregate_dir)
-    except ValueError as exc:
-        raise InputError(
-            "A nested item is a reference to a path that is not within the directory of its "
-            f"parent. {item=!r}, expected parent path: {aggregate_dir!r}"
-        ) from exc
+    if item.hidden and item_reference_type == ItemReferenceType.DIR:
+        raise InputError(f"A hidden item is a directory. {item=!r}")
 
-    # Check that the item is directly within the current directory
-    if len(item_to_aggregate_path.parents) != 1:
-        raise InputError(
-            "A nested item is a reference to a path that is not immediately within the "
-            f"directory of its parent. {item=!r}, expected parent path: {aggregate_dir!r}"
-        )
+    if item_reference_type in {ItemReferenceType.DIR, ItemReferenceType.FILE}:
```

Comment:
Out of curiosity, may I ask why a set is used rather than a tuple here please?

Reply:
In practice the difference is negligible - the reason is that checking for whether a value is in a set is `O(1)` and checking whether it is in a tuple is `O(N)`. In this case N is 2 - its just that there isn't really a cost to using a set so why wouldn't a set be used?